### DISCUSSION TITLE
Should we consider a new design to no algorithm engineer modification needed?

### DISCUSSION THREAD

**Discussion by joshua-xia** [microsoft/nni#3792] [Ideas]
**What would you like to be added**:
a new design let algorithm engineer need NOT modify the code according to the parameters input

**Why is this needed**:
reduce algorithm engineer 's work

**Without this feature, how does current nni work**：
current nni need modify the code according to the parameters

**Components that may involve changes**:
the pipeline design

**Brief description of your proposal if any**:
Not yet decide

**Comment by xuehui1991** [comment]
Hello @joshua-xia,  could you give more detail about your proposal? Could you give some examples? Thank you.

**Comment by joshua-xia** [comment]
I am trying to write a prototype. you know to use the current NNI, the algorithm engineer have to modify the model code according to the parameters input,  and report the results using the SDK API, I wonder if there is a design/implement reduce this kind of modification

**Comment by joshua-xia** [comment]
for example:

I have to write the following code to get model:

def get_model(PARAMS):
    '''Get model according to parameters'''
    if PARAMS.get('_name') == 'GradientBoostingClassifier':
        model = ensemble.GradientBoostingClassifier()
        model.loss = PARAMS.get('loss')
        model.learning_rate = PARAMS.get('learning_rate')
        model.n_estimators = PARAMS.get('n_estimators')
        model.criterion = PARAMS.get('criterion')
    elif PARAMS.get('_name') == 'SVM':
        import sklearn.svm.SVC
        model = sklearn.svm.SVC()
        model.C = PARAMS.get('C')
        model.keral = PARAMS.get('keral')
        model.degree = PARAMS.get('degree')
        model.gamma = PARAMS.get('gamma')
        model.coef0 = PARAMS.get('coef0')
    elif PARAMS.get('_name') == 'RF':
....

Can we provide API in NNI SDK such as:  nni.get_next_model() to get model directly?

**Comment by scarlett2018** [comment]
Thanks @joshua-xia  for raising the feature request. This is not in our recent roadmap, we will leave this open in case any other users have the interests or raise any proposals.
